### PR TITLE
For #14619: Show half a tab top offset when scrolling to selected tab.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
This seems stuck. Closing and reopening.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt [hunk 1]
```diff
@@ -659,6 +659,10 @@ class TabTrayView(
                 1
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:659]: I am wondering if we can remove this "1" that I added for displaying the tab item above the selected browser tab.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt [hunk 2]
```diff
@@ -659,6 +659,10 @@ class TabTrayView(
...
 
             layoutManager?.scrollToPosition(recyclerViewIndex)
+            smoothScrollBy(
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:662]: We should avoid scrolling if the first item is selected.

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt (hunk 2 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:660]: I have reversed the scrolling and removed the `+1`.
So, we scroll to the selected item, then scroll "down" so half of the item above will be visible. If there is no item above, the scroll would not happen (there is not scroll space left).

**Review by gabrielluong — APPROVED** [review]
Changes look good. Just waiting for a ux review.

**Comment by mcarare** [comment]
@topotropic can you review this from the UX side? Thank you!

**Review by topotropic — COMMENTED** [review]
Thanks! Apologies for not getting back to the comment in the original issue that the x looks like and arrow when you cut it in half (https://github.com/mozilla-mobile/fenix/issues/14619#issuecomment-710645863). I just checked out the implementation and it looks like that when you are on tab 1, 2 or 3 we cut in half but once you are on the 4th tab we show 3/4 or so of the tab. Is that correct?

**Comment by mcarare** [comment]
@topotropic No, we always show 1/2. When the selected tab is further down the list, it depends on the total number of items, as the selected tab could not be shown at the top if there are not sufficient items below it to scroll.
So, what is the proposed solution for the arrow issue?

**Review by topotropic — APPROVED** [review]
Looks good –  with the 2lines we either have the choice to roughly show 1/3 of the tab which would hide the URL when you have a 1line title or showing 2/3 of the tab which might be too much. The implemented solution is fine. Thank you!

**Pull request merged**